tests/func/experiments/test_utils.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+from funcy import first
+
+from dvc.repo.experiments.utils import exp_refs_by_rev, resolve_exp_ref
+
+
+@pytest.mark.parametrize(
+    "full_name, test_remote", [(True, False), (False, True), (False, False)]
+)
+def test_remove_remote(
+    tmp_dir, scm, dvc, exp_stage, git_upstream, full_name, test_remote
+):
+    remote = None
+    results = dvc.experiments.run(exp_stage.addressing, params=["foo=2"])
+    exp = first(results)
+    ref_info = first(exp_refs_by_rev(scm, exp))
+    if test_remote:
+        remote = git_upstream.url
+        dvc.experiments.push(remote, ref_info.name)
+
+    if full_name:
+        exp_name = str(ref_info)
+    else:
+        exp_name = ref_info.name
+
+    assert resolve_exp_ref(scm, exp_name, remote).name == ref_info.name
```

Comment:
```suggestion
    assert str(resolve_exp_ref(scm, exp_name, remote)) == str(ref_info)
```

We should be checking that we get the correct ref entirely, not just the name segment.

Actually it would be even better to just define `__hash___()` properly for ExpRefInfo's so that we can just do `if ref_info1 == ref_info2`, but it doesn't have to be done in this PR

Reply:
Also I think this should really be in it's own unit test (it probably also belongs in `tests/unit/experiments/test_utils.py` since it's not a func test). And it should be tested using SCM only, without a full DVC repo instance. Basically something like

```py
@pytest.mark.parametrize("name_only", [True, False])
def test_resolve_exp_ref(tmp_dir, scm, name_only):
    tmp_dir.scm_gen("foo", "foo", commit="init")
    rev = scm.get_rev()
    ref = "/".join([EXPS_NAMESPACE, "ab", "c123", "foo"])
    scm.gitpython.set_ref(ref, rev)
    ref_info = resolve_exp_ref(scm, "foo" if name_only else ref)
    assert isinstance(ref_info, ExpRefInfo)
    assert str(ref_info) == ref
```

Ideally it should also be parametrized for remotes, to do the setup for testing a remote ref you just need
```
scm.push_refspec(remote, ref, ref)
```
(see https://github.com/iterative/dvc/blob/master/tests/unit/scm/test_git.py#L231 for examples on how to set up the git remote)